### PR TITLE
[SPARK-25972][PYTHON] Missed JSON options in streaming.py

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

Added JSON options for `json()` in streaming.py that are presented in the similar method in readwriter.py. In particular, missed options are `dropFieldIfAllNull` and `encoding`.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@HyukjinKwon Thank you for pointing out the missed options. Please, have a look at the PR.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Oh, let's name it `[PYTHON]` BTW.

**Comment by MaxGekk** [comment]
> Oh, let's name it [PYTHON] BTW.

I see some of on-going PR have tag `[PYSPARK]`. Is this somewhere documented?

**Comment by HyukjinKwon** [comment]
Yea .. actually that's documented in https://spark.apache.org/contributing.html . Strictly it should be` PYTHON`

> The PR title should be of the form [SPARK-xxxx][COMPONENT] Title, where SPARK-xxxx is the relevant JIRA number, COMPONENT is one of the PR categories shown at spark-prs.appspot.com and Title may be the JIRA’s title or a more specific title describing the PR itself.

screen shot 2018-11-08 at 7 19 07 pm

I tried to be clear for myself so I took a look before but .. not a big deal at all :D.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/streaming.py [hunk 1]
```diff
@@ -467,11 +468,18 @@ def json(self, path, schema=None, primitivesAsString=None, prefersDecimal=None,
...
+                         the encoding of input JSON will be detected automatically
+                         when the multiLine option is set to ``true``.
         :param lineSep: defines the line separator that should be used for parsing. If None is
                         set, it covers all ``\\r``, ``\\r\\n`` and ``\\n``.
         :param locale: sets a locale as language tag in IETF BCP 47 format. If None is set,
                        it uses the default value, ``en-US``. For instance, ``locale`` is used while
                        parsing dates and timestamps.
+        :param dropFieldIfAllNull: whether to ignore column of all null values or empty
+                                   array/struct during schema inference. If None is set, it
+                                   uses the default value, ``false``.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/streaming.py]: @MaxGekk, let's match its order (the doc and parameters).

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**